workflows/eager.nf

```diff
@@ -333,7 +343,27 @@ workflow EAGER {
         ch_versions = ch_versions.mix( PRESEQ_LCEXTRAP.out.versions )
     }
 
-     //
+
+    //
+    // MODULE: Bedtools coverage 
+    //
+    
+    if ( params.run_bedtools_coverage ) {
+        ch_dedupped_bams.combine(ch_anno_for_bedtools).map{
+            meta, bam, bai, anno ->
+            [meta, anno, bam]
+        }.set { ch_dedupped_for_bedtools }
+
+        // Running samtools view to get header
+        SAMTOOLS_VIEW_GENOME(ch_dedupped_bams, [], [])
+        BEDTOOLS_COVERAGE(ch_dedupped_for_bedtools, SAMTOOLS_VIEW_GENOME.out.sam.map{[it[1]]})
+        BEDTOOLS_COVERAGE_MEAN(ch_dedupped_for_bedtools, SAMTOOLS_VIEW_GENOME.out.sam.map{[it[1]]})
+        ch_versions = ch_versions.mix( SAMTOOLS_VIEW_GENOME.out.versions )
+        ch_versions = ch_versions.mix( BEDTOOLS_COVERAGE.out.versions )
```

Comment:
Missing `BEDTOOLS_COVERAGE_MEAN` mixing (we want a 1-to-1 relationship ebtween executing modules and the list of verisons in the table)

Reply:
What does the ch_versions do? I assumed it was just to keep track of the tools used. So if bedtools will be accounted for with ch_versions = ch_versions.mix( BEDTOOLS_COVERAGE.out.versions ). What would 
ch_versions = ch_versions.mix( BEDTOOLS_COVERAGE_MEAN.out.versions ) add?

Reply:
It gathers all the `version.yml` files, to eventually make a long list of _nextflow module (i.e. process)_ and their subsequent versions. It can be that you have two versions of the same tool/module in the same piepline that would report different versions.